source/time_stepping/convection_time_step.cc

```diff
@@ -51,11 +57,36 @@ namespace aspect
             fe_values.reinit (cell);
             fe_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
                                                                                         velocity_values);
+            in.reinit(fe_values, cell, this->introspection(), this->get_solution());
+            this->get_material_model().fill_additional_material_model_inputs(in, this->get_solution(), fe_values, this->introspection());
+            this->get_material_model().evaluate(in, out);
 
             double max_local_velocity = 0;
             for (unsigned int q=0; q<n_q_points; ++q)
-              max_local_velocity = std::max (max_local_velocity,
-                                             velocity_values[q].norm());
+              {
+                if (this->introspection().compositional_name_exists("porosity"))
+                  {
+                    const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
```

Comment:
move this to the top of the file, where you also compute `consider_darcy_timestep`

Reply:
Unless I'm misunderstanding your comments, I don't know if it's possible to not compute the porosity_idx here because you need to divide by the porosity to calculate the Darcy velocity

Reply:
But you can compute the porosirty_idx above, and use it here. See my comment above.